pytest_minio_mock/plugin.py

```diff
@@ -446,9 +481,24 @@ def put_object(
                 bucket_name=bucket_name,
                 object_name=None,
             )
-        self.buckets[bucket_name][object_name] = MockMinioObject(
-            object_name=object_name, data=data
-        )
+        if self.get_bucket_versioning(bucket_name).status == ENABLED:
```

Comment:
I would do it this way to simplify the code:
        
        version = str(uuid4())
        obj = MockMinioObject(object_name=object_name, data=data, version_id=version)
        if self.get_bucket_versioning(bucket_name).status != DISABLED:
            self.buckets[bucket_name][object_name] = {}   # delete other versions
        self.buckets[bucket_name][object_name][version] = obj

That simplifies the code significantly... Similar changes need to happen in get_object()